src/kohlrahbi/unfoldedahb/unfoldedahbtable.py

```diff
@@ -286,12 +287,31 @@ def dump_flatahb_json(self, output_directory_path: Path) -> None:
 
         flatahb_output_directory_path = output_directory_path / str(edifact_format) / "flatahb"
         flatahb_output_directory_path.mkdir(parents=True, exist_ok=True)
-
-        dump_data = FlatAnwendungshandbuchSchema().dump(self.convert_to_flat_ahb())
-
-        with open(
-            flatahb_output_directory_path / f"{self.meta_data.pruefidentifikator}.json", "w", encoding="utf-8"
-        ) as file:
+        flat_ahb = self.convert_to_flat_ahb()
+
+        file_path = flatahb_output_directory_path / f"{self.meta_data.pruefidentifikator}.json"
+        if file_path.exists():
+            with open(file_path, "r", encoding="utf-8") as file:
+                existing_flat_ahb = FlatAnwendungshandbuchSchema().load(json.load(file))
+        dump_equals_existing_file_except_for_guids = flat_ahb.meta == existing_flat_ahb.meta and len(
+            flat_ahb.lines
+        ) == len(existing_flat_ahb.lines)
+        if dump_equals_existing_file_except_for_guids:
+            ahb_line_schema = AhbLineSchema()
+            for line, existing_line in zip(flat_ahb.lines, existing_flat_ahb.lines):
+                line_copy = ahb_line_schema.load(ahb_line_schema.dump(line))
```

Comment:
This is a C# pattern for deep copies isn't it?
I thought in Python there are methods to create deep copies.
* asking chatGPT *
ah yes there is a standard lib called `copy`
See https://chat.openai.com/share/4ad4c1ad-6f13-4a21-a0e3-aaf75e17c556